packages/make-styles/src/ax.ts

```diff
@@ -1,57 +1,57 @@
+import { DEFINITION_LOOKUP_TABLE, HASH_LENGTH, RTL_PREFIX, SEQUENCE_PREFIX } from './constants';
 import { hashString } from './runtime/utils/hashString';
 import { MakeStylesMatchedDefinitions } from './types';
 
 // Contains a mapping of previously resolved sequences of atomic classnames
-const axResults: Record<string, string> = {};
+const axCachedResults: Record<string, string> = {};
+
+const SEQUENCE_SIZE = SEQUENCE_PREFIX.length + HASH_LENGTH;
 
 /**
  * Function can take any number of arguments, joins classes together and deduplicates atomic declarations generated by
- * `makeStyles()`.
+ * `makeStyles()`. Handles scoped directional styles.
  *
  * Classnames can be of any length, this function can take both atomic declarations and class names.
  *
  * Input:
  * ```
  * // not real classes
- * ax('ui-button', 'displayflex', 'displaygrid')
+ * ax('ltr', ['ui-button', 'displayflex', 'displaygrid'])
+ * ax('rtl', ['ui-button', 'displayflex', 'displaygrid'])
  * ```
  *
  * Output:
  * ```
  * 'ui-button displaygrid'
+ * 'rtl ui-button displaygrid'
  * ```
  */
-export function ax(...classNames: (string | false | undefined)[]): string;
-
-export function ax(): string {
-  // arguments are parsed manually to avoid double loops as TS & Babel transforms rest via an additional loop
-  // @see https://babeljs.io/docs/en/babel-plugin-transform-parameters
+export function ax(dir: 'ltr' | 'rtl', classNames: (string | false | undefined)[]): string {
+  const isRtl = dir === 'rtl';
 
   let resultClassName = '';
   // Is used as a cache key to avoid object merging
   let sequenceMatch = '';
 
   const sequenceMappings: MakeStylesMatchedDefinitions[] = [];
 
-  for (let i = 0; i < arguments.length; i++) {
-    // eslint-disable-next-line prefer-rest-params
-    const argument = arguments[i];
+  for (let i = 0; i < classNames.length; i++) {
+    const className = classNames[i];
 
-    if (typeof argument === 'string') {
+    if (typeof className === 'string') {
       // All classes generated by `makeStyles()` are prefixed by a sequence hash, this allows to identify class sets
       // without parsing each className in a string
-      const sequenceIndex = argument.indexOf(SEQUENCE_PREFIX);
+      const sequenceIndex = className.indexOf(SEQUENCE_PREFIX);
 
       if (sequenceIndex === -1) {
-        resultClassName += argument + ' ';
+        resultClassName += className + ' ';
       } else {
-        const sequenceId = argument.slice(sequenceIndex, sequenceIndex + 9);
+        const sequenceId = className.slice(sequenceIndex, sequenceIndex + SEQUENCE_SIZE);
         const sequenceMapping = DEFINITION_LOOKUP_TABLE[sequenceId];
 
         // Handles a case with mixed classnames, i.e. "ui-button ATOMIC_CLASSES"
         if (sequenceIndex > 0) {
-          resultClassName += argument.slice(0, sequenceIndex);
+          resultClassName += className.slice(0, sequenceIndex);
         }
 
         if (sequenceMapping) {
```

Comment:
If `sequenceMapping` is not found, the sequence classname and all subsequent classes in the string are thrown away. Is it expected? If so, could you add a UT to describe this behaviour?

Reply:
> Is it expected? If so, could you add a UT to describe this behaviour?

Yes, I added a warning and a unit test for it, 7a4beea.